src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -206,20 +222,64 @@ public bool IsRenameOverloadsEditable
         public bool IsRenameOverloadsVisible
             => Session.HasRenameOverloads;
 
+        public bool AllowUserInput
+            => !Session.IsCommitInProgress;
+
+        private Visibility _visibility;
+        public Visibility Visibility
+        {
+            get => _visibility;
+            set
+            {
+                if (value != _visibility)
+                {
+                    Set(ref _visibility, value);
+                    NotifyPropertyChanged();
+                }
+            }
+        }
+
         public TextSpan StartingSelection { get; }
 
         public bool Submit()
         {
+            using var token = _asyncListener.BeginAsyncOperation(nameof(Submit));
             if (StatusSeverity == Severity.Error)
             {
                 return false;
             }
 
             SmartRenameViewModel?.Commit(IdentifierText);
-            Session.Commit();
+            _ = CommitAsync().ReportNonFatalErrorAsync();
             return true;
         }
 
+        private async Task CommitAsync()
+        {
+            try
+            {
+                await Session.CommitAsync(previewChanges: false).ReportNonFatalErrorAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex) when (FatalError.ReportAndCatch(ex, ErrorSeverity.Critical))
+            {
+                // Show a nice error to the user via an info bar
+                var errorReportingService = Session.Workspace.Services.GetService<IErrorReportingService>();
```

Comment:
there was a CA(false) above, but i think IErrorService is thread affinitized.  Likely needs to be CA(true), or an explicit SwithcToMainThread here.